Sources/demo/main.swift

```diff
@@ -0,0 +1,343 @@
+            {
+                print("[Provisioning] Assigned Hub: \(provResponse.RegistrationState.AssignedHubHostname)")
+                isDeviceProvisioned = true;
+                
+                assignedHub = provResponse.RegistrationState.AssignedHubHostname
+                assignedDeviceID = provResponse.RegistrationState.DeviceID
+            }
+            
+        case let packet as SubAckPacket:
+            print("[Provisioning] Suback Received: \(packet)");
+
+        default:
+            print("[Provisioning] Packet Received: \(packet)")
+        }
+    }
+
+    func mqttClient(_: MQTTClient, didChange state: ConnectionState) {
+        if state == .disconnected {
+            print("[Provisioning] MQTT state:\(state)")
+        }
+    }
+
+    func mqttClient(_: MQTTClient, didCatchError error: Error) {
+        print("[Provisioning] Error: \(error)")
+    }
+
+    public func connectToProvisioning() {
+        print("[Provisioning] Connecting to Provisioning")
+        mqttClient.connect()
+    }
+    
+    public func disconnectFromProvisioning() {
+        print("[Provisioning] Disconnecting from Provisioning")
+        mqttClient.disconnect()
+    }
+
+    public func subscribeToAzureDeviceProvisioningFeature() {
+        print("[Provisioning] Subscribing to Provisioning")
+        let deviceProvisioningTopic = AzProvClient.GetDeviceProvisioningSubscribeTopic()
+        print("[Provisioning] Subscribing to topic: \(deviceProvisioningTopic)")
+        mqttClient.subscribe(topic: deviceProvisioningTopic, qos: QOS.1)
+    }
+
+    public func sendDeviceProvisioningRequest() {
+        print("[Provisioning] Requesting to be Provisioned")
+        let deviceProvisioningRequestTopic = AzProvClient.GetDeviceProvisioningRegistrationPublishTopic()
+        mqttClient.publish(topic: deviceProvisioningRequestTopic, retain: false, qos: QOS.1, payload: "")
+    }
+
+    public func sendDeviceProvisioningPollingRequest(operationID: String) {
+        print("[Provisioning] Querying Provisioning")
+        let deviceProvisioningQueryTopic = AzProvClient.GetDeviceProvisioningQueryTopic(operationID: operationID)
+        mqttClient.publish(topic: deviceProvisioningQueryTopic, retain: false, qos: QOS.1, payload: "")
+    }
+}
+
+class DemoHubClient: MQTTClientDelegate {
+
+    /// Azure IoT Client
+    private var AzHubClient : AzureIoTHubClient! = nil
+
+    /// MQTT Client
+    private var mqttClient: MQTTClient! = nil
+    
+    var delegateDispatchQueue: DispatchQueue {
+        queue
+    }
+
+    public init(iothub: String, deviceId: String)
+    {
+        AzHubClient = AzureIoTHubClient(iothubUrl: iothub, deviceId: deviceId)
+
+        let caCert: [UInt8] = Array(myBaltimore.utf8)
+        let clientCert: [UInt8] = Array(myCert.utf8)
+        let keyCert: [UInt8] = Array(myCertKey.utf8)
+
+        let tlsConfiguration = try! TLSConfiguration.forClient(minimumTLSVersion: .tlsv11,
```

Comment:
We could use makeConfiguration, since forClient is deprecated and generates a warning.
Example:

```swift
 var tlsConfiguration = TLSConfiguration.makeClientConfiguration()
 tlsConfiguration.minimumTLSVersion = .tlsv11
 tlsConfiguration.maximumTLSVersion = .tlsv12
 tlsConfiguration.trustRoots = try! NIOSSLTrustRoots.certificates(NIOSSLCertificate.fromPEMFile(caCert))
 tlsConfiguration.certificateVerification = .noHostnameVerification
```

Reply:
Yea I'll tackle this in a bit when I can test it.